test/test_type_traits.cpp

```diff
@@ -1320,4 +1327,9 @@ namespace
     CHECK_FALSE(bool(etl::is_base_of_all<Base, D1, D2, D3, D4>::value));
 #endif
   }
+
+  //*************************************************************************
+  TEST(test_type_identity) {
+    CHECK(type_identity_test_add(1.5f, 2) == 3.5f);
```

Comment:
You should never test floating point numbers for equality.
Use `CHECK_CLOSE` with a small tolerance value, such as `0.01f`.

Reply:
Why not? I used 0.5 as the fraction, which is a negative power of two, so no rounding is involved.

Reply:
It's just standard good practice.
It protects against code breaking if someone were to change a value in the future without realising the implications.